sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/RetryPolicy.java

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp;
+
+import com.azure.core.amqp.exception.AmqpException;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeoutException;
+
+import static com.azure.core.amqp.exception.ErrorCondition.SERVER_BUSY_ERROR;
+
+/**
+ * An abstract representation of a policy to govern retrying of messaging operations.
+ */
+public abstract class RetryPolicy implements Cloneable {
```

Comment:
These retry policies are fairly generic and not specific to amqp (except the exception thrown). Can this be moved to azure-core instead of azure-core-amqp?

Reply:
Unfortunately, our retry is different the Retry in azure-core in that the one in azure core reads from the retry after header, or reads the appropriate HTTP status code then then attempts to retry.